packages/ui/src/council/modals/AnnounceCandidate/machine.ts

```diff
@@ -0,0 +1,130 @@
+import BN from 'bn.js'
+import { assign, createMachine } from 'xstate'
+
+import { Account } from '@/accounts/types'
+import { EmptyObject } from '@/common/types'
+import { SetAccountEvent, SetAmountEvent, SetTitleEvent } from '@/proposals/modals/AddNewProposal/machine'
+
+interface StakingContext {
+  stakingAccount?: Account
+  stakingAmount?: BN
+}
+
+interface RewardAccountContext extends Required<StakingContext> {
+  rewardAccount?: Account
+}
+
+export type AnnounceCandidateContext = Partial<StakingContext & RewardAccountContext>
+
+export type AnnounceCandidateState =
+  | { value: 'requirementsVerification'; context: EmptyObject }
+  | { value: 'requirementsFailed'; context: EmptyObject }
+  | { value: 'requiredStakeVerification'; context: EmptyObject }
+  | { value: 'requiredStakeFailed'; context: EmptyObject }
+  | { value: 'staking'; context: Required<StakingContext> }
+  | { value: 'rewardAccount'; context: Required<RewardAccountContext> }
+  | { value: 'candidateProfile'; context: Required<RewardAccountContext> }
+  | { value: 'candidateProfile.titleAndDescription'; context: Required<RewardAccountContext> }
+  | { value: 'candidateProfile.summaryAndBanner'; context: Required<RewardAccountContext> }
+  | { value: 'candidateProfile.finishCandidateProfile'; context: Required<RewardAccountContext> }
+  | { value: 'success'; context: Required<RewardAccountContext> }
+  | { value: 'error'; context: AnnounceCandidateContext }
+
+export type AnnounceCandidateEvent =
+  | { type: 'FAIL' }
+  | { type: 'BACK' }
+  | { type: 'NEXT' }
+  | SetAccountEvent
+  | SetAmountEvent
+  | SetTitleEvent
```

Comment:
I think that for now, we could revert this and inline those. Just to not mix machines for now.